### PR TITLE
Graphql token

### PR DESCRIPTION
**Pull request by JanuszJakubiec**
Adding graphql resolvers for mod_auth_token.

### CODE REVIEW

File: src/mod_auth_token_api.erl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    case mongoose_domain_api:get_domain_host_type(LServer) of
+        {ok, HostType} ->
+            try mod_auth_token:revoke(HostType, Jid) of
+                not_found ->
+                    {not_found, "User or token not found."};
+                ok ->
+                    {ok, "Revoked."};
+                error ->
+                    {internal_server_error, "Internal server error."}
+            catch _:_ ->
```
Comment (arcusfelis) [inline src/mod_auth_token_api.erl:25]: Log the error here

File: src/mod_auth_token_api.erl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    {internal_server_error, "Internal server error."}
+            end;
+        _ ->
```
Comment (arcusfelis) [inline src/mod_auth_token_api.erl:28]: it should be "Unknown domain"

File: src/mod_auth_token_api.erl [hunk 3]
```diff
@@ -0,0 +1,53 @@
+-module(mod_auth_token_api).
+
+-include("jlib.hrl").
+-include("mod_auth_token.hrl").
+
+-export([revoke_token_command/1, create_token/1]).
+
+-import(mod_auth_token, [token/3, serialize/1]).
+
+-spec revoke_token_command(User) -> ResTuple when
```
Comment (arcusfelis) [inline src/mod_auth_token_api.erl:10]: We usually use `{ok, Data}` or `{error, Reason}`, where Reason could be `{not_found, "User or token not found"}`. Not super critical, just a convention

**Review by arcusfelis — COMMENTED** [review]
added some comments

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/130563) / small_tests / bfc72f91c1ae378cd22648bd0f1b9ff7d8ffc177
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3713/130563/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3713/130563/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-07-25_07.05.02/lib.mongooseim.logs/run.2022-07-25_07.05.02/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/130612) / small_tests / dbca0ed63c4f977fe73a3d125da013a446a524bf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3713/130612/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3713/130612/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-07-25_07.41.20/lib.mongooseim.logs/run.2022-07-25_07.41.20/suite.log.html)

File: src/graphql/user/mongoose_graphql_token_user_mutation.erl [hunk 4]
```diff
@@ -0,0 +1,33 @@
+-module(mongoose_graphql_token_user_mutation).
+ -behaviour(mongoose_graphql).
+
+ -export([execute/4]).
+
+ -ignore_xref([execute/4]).
+
+ -include("../mongoose_graphql_types.hrl").
+
+ -import(mongoose_graphql_helper, [null_to_default/2, make_error/2]).
```
Comment (arcusfelis) [inline src/graphql/user/mongoose_graphql_token_user_mutation.erl:10]: null_to_default is not used

**Review by arcusfelis — COMMENTED** [review]
okk, just add error reporting after catch-all (I think the error would be logged anyway centrally, but yea, good idea to still report after catching all, because you loose reason and stacktrace in the code you have now) and check if imported functions are actually used

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/130642) / small_tests / 95d8c489c499e913a17fa3d6c169bcc7743f837f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3713/130642/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3713/130642/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-07-25_10.17.18/lib.mongooseim.logs/run.2022-07-25_10.17.19/suite.log.html)

**Review by arcusfelis — APPROVED** [review]
ok

**Pull request merged**